pkg/testutil/framework.go

```diff
@@ -307,6 +315,182 @@ func (f *Framework) SetupServiceCluster(ctx context.Context, cl *RecordingClient
 	return serviceCluster
 }
 
+func (f *Framework) KubeCarrierOperatorCheck(
+	ctx context.Context,
+	t *testing.T,
+	managementClient *RecordingClient,
+	managementScheme *runtime.Scheme,
+) {
+	namePrefix := constants.KubeCarrierDefaultName + "-operator"
+	namespaceName := constants.KubeCarrierDefaultNamespace
+
+	f.componentCheck(ctx, managementClient, managementScheme, t, namePrefix, []runtime.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-manager", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Webhook Service
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-webhook-service", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// ClusterRoleBinding
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-rolebinding", namePrefix),
+			},
+		},
+		// ClusterRole
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-role", namePrefix),
+			},
+		},
+		// RoleBinding
+		&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-rolebinding", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Role
+		&rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-role", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "accounts.catalog.kubecarrier.io",
+			},
+		},
+	})
+}
+
+func (f *Framework) E2EOperatorCheck(
+	ctx context.Context,
+	t *testing.T,
+	serviceClient *RecordingClient,
+	serviceScheme *runtime.Scheme,
+) {
+	namePrefix := "e2e-operator"
+	namespaceName := "kubecarrier-e2e-operator"
+
+	f.componentCheck(ctx, serviceClient, serviceScheme, t, namePrefix, []runtime.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-manager", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// ClusterRoleBinding
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-rolebinding", namePrefix),
+			},
+		},
+		// ClusterRole
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-role", namePrefix),
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "dbs.fake.kubecarrier.io",
+			},
+		},
+	})
+}
+
+func (f *Framework) KubeCarrierCheck(
```

Comment:
This function doesn't utilize f Framework as pointer receiver (if we ignore `componentCheck` which doesn't utilize f at all). Thus it can be a free-standing function.

Reply:
For inline `KubeCarrierCheck` and other check functions:
I keep them in the framework package, because they will be invoked multiple times once we have more service clusters, and kubecarrier installations in our e2e tests, and I didn't see much benefits to inline them so I just keep them in the framework package, and we can reuse them later.